contracts/ARC20.sol

```diff
@@ -0,0 +1,219 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity >=0.7.0;
+
+library NativeAssets {
+    address constant balanceAddr = 0x0100000000000000000000000000000000000001;
+    address constant transferAddr = 0x0100000000000000000000000000000000000002;
+
+    function assetBalance(address addr, uint256 assetID) public returns (uint256) {
+        (bool success, bytes memory data) = balanceAddr.call(abi.encodePacked(addr, assetID));
+        require(success, "assetBalance failed");
+        return abi.decode(data, (uint256));
+    }
+    
+    function getBalance(uint256 assetID) public returns (uint256) {
+        return assetBalance(msg.sender, assetID);
+    }
+    
+    function assetCall(address addr, uint256 assetID, uint256 assetAmount, bytes memory callData) public returns (bytes memory) {
+        (bool success, bytes memory data) = transferAddr.call(abi.encodePacked(addr, assetID, assetAmount, callData));
+        require(success, "assetCall failed");
+        return data;
+    }
+    
+    function transferAsset(address addr, uint256 assetID, uint256 assetAmount) public returns (bytes memory) {
+        return assetCall(addr, assetID, assetAmount, "");
+    }
+}
+
+contract ARC20 {
```

Comment:
Did you borrow/adapt this source from anywhere? Do we need attribution? Is the license compatible?

Reply:
Does our implementation of the core ERC20-bits differ from existing implementations such as Open Zeppelin's? Also, should we consider extending Open Zeppelin's lib and only implement the ARC20-specific parts? Or do we not want to be that tightly coupled to Open Zeppelin's implementation?

```sol
import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
contract ARC20 is ERC20 {
```

Reply:
I would recommend going the OpenZeppelin route if we can. I wasn't sure if you copy/pasted any code from them. If we did, we need to add an MIT license to the whole repo. We're GPLv3 rn

Reply:
@dasconnor do you have any examples of overlaying a small amount of extra functionality on top of an existing ERC-20 via OpenZeppelin?